youtube_dl/extractor/tiktok.py

```diff
@@ -105,34 +148,75 @@ def _real_initialize(self):
             'https://www.tiktok.com/', None, note='Setting up session')
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
-        webpage = self._download_webpage(url, video_id)
-        page_props = self._parse_json(self._search_regex(
-            r'<script[^>]+\bid=["\']__NEXT_DATA__[^>]+>\s*({.+?})\s*</script',
-            webpage, 'data'), video_id)['props']['pageProps']
+        m = re.match(self._VALID_URL, url).groupdict()
+        video_id = m['id']
+        if 'user_id' in m:
+            url = 'https://www.tiktok.com/@%(user_id)s/video/%(id)s/' % m
+
+        webpage = self._download_webpage(url.replace('://m.', '://www.'), video_id)
+
+        page_props = self._get_SIGI_STATE(video_id, webpage)
+        if try_get(page_props, lambda x: x['ItemModule'][video_id]['video'], dict):
+            data = page_props['ItemModule'][video_id]
+            if data.get('privateItem'):
+                raise ExtractorError('This video is private', expected=True)
+            return self._extract_video(data, video_id)
+
+        # old page format - may still be served?
+        page_props = self._parse_json(
+            get_element_by_id('__NEXT_DATA__', webpage),
+            video_id)['props']['pageProps']
         data = try_get(page_props, lambda x: x['itemInfo']['itemStruct'], dict)
         if not data and page_props.get('statusCode') == 10216:
             raise ExtractorError('This video is private', expected=True)
         return self._extract_video(data, video_id)
 
 
-class TikTokUserIE(TikTokBaseIE):
-    _VALID_URL = r'https://(?:www\.)?tiktok\.com/@(?P<id>[^/?#&]+)'
+class TikTokUserIE(TikTokIE):
+    IE_DESC = 'TikTok user profile extractor'
+    IE_NAME = 'tiktok:user'
+    _VALID_URL = r'https?://(?:(?:www|m)\.)?tiktok\.com/@(?P<id>[^/?#&]+)'
     _TESTS = [{
         'url': 'https://www.tiktok.com/@zureeal',
         'info_dict': {
             'id': '188294915489964032',
         },
-        'playlist_mincount': 24,
+        'playlist_mincount': 30,
+        'expected_warnings': [
+            'More videos are available',
+        ],
     }]
-    _WORKING = False
 
     @classmethod
     def suitable(cls, url):
-        return False if TikTokIE.suitable(url) else super(TikTokUserIE, cls).suitable(url)
+        return False if TikTokIE.suitable(url) else super(TikTokBaseIE, cls).suitable(url)
 
     def _real_extract(self, url):
         user_id = self._match_id(url)
+
+        webpage = self._download_webpage(url.replace('://m.', '://www.'), user_id)
+
+        page_props = self._get_SIGI_STATE(user_id, webpage)
+        user_data = try_get(page_props, lambda x: x['UserModule']['users'], dict)
+        if user_data:
```

Comment:
It should be
```py
        if not user_data:
            raise ExtractorError(...)
        ...
```
If the extractor returns `None`, youtube-dl will just silently exit. See https://github.com/yt-dlp/yt-dlp/issues/3776#issuecomment-1155586954